### PR TITLE
Add 'Mbed-OS platfom overview' description to docs directory.

### PR DESCRIPTION
**Pull request by lukgni**
## Problem
We decided to add general Mbed platform overview as an entry level documentation for any new person who want to use CHIP+Mbed.

Proposed documents consists some general information about Mbed, used network stacks, supported examples and CHIP integration, etc.
Probably section about IP stack integration will need to be changed by a person who know more about Mbed IP integration.

## Change set
Adding mbed_os_platform_overview.ms file
Add images used in above file to docs/images directory.

### CODE REVIEW

File: docs/guides/mbedos_platform_overview.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+For Wi-FI communication purposes, the Mbed-OS application is
+using internal Mbed LWIP stack and Mbed Socket API together. To integrate it with CHIP
+network layer, special glue socket layer has been introduced to take care of
+adapting the Mbed socket to BSD interface which is used inside the CHIP endpoints
+implementation.
+
+## CHIP integration
+The Bluetooth LE and Wi-Fi used stacks provided
+by the Mbed-OS has been integrated with the CHIP stack using a
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:72]:
```suggestion
by the Mbed-OS have been integrated with the CHIP stack using a
```

File: docs/guides/mbedos_platform_overview.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+special intermediate layer.
+
+This layer contains platform-specific implementations of abstract
+manager interfaces defined in the CHIP stack. The application is able to use CHIP's platform
+agnostic interfaces and no additional platform-related actions are needed to
+perform communication through the CHIP stack.
+
+# CHIP example applications
+The ARM Mbed-OS platform provides bunch of ported CHIP sample applications which can be used
+to spped-up developemnt process:
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:82]:
```suggestion
Matter application sample are provided for the Mbed OS platform. They can be used to speed up development: 
```

File: docs/guides/mbedos_platform_overview.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- shell
+- all-clusters-app
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:84]: Please add links for these applications.

File: docs/guides/mbedos_platform_overview.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+- [lock-app](../../examples/lock-app/mbed/README.md)
+- [lighting-app](../../examples/lighting-app/mbed/README.md)
+
+## CHIP stack configuration
+In each of supported examples, the CHIP stack can be configured by modyfying `CHIPProjectConfig.h` file
+which is placed inside the project directory.
+
+## Mbed-OS configuration
+The Mbed-OS gives possibility to tweak its parameters by using
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:93]:
```suggestion
Mbed-OS gives possibility to tweak its parameters by using the 
```

File: docs/guides/mbedos_platform_overview.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+[Mbed-OS configuration system](https://os.mbed.com/docs/mbed-os/v6.12/program-setup/advanced-configuration.html).
+The same configuration system can be used to change default hardware target setup used in
+application. Additionally, it is the first place for adding support of the new hardware
+target support into the application. Mbed-OS configuration system can be accessed by modyfing
+`mbed_app.json` file which exists in each sample project directory.
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:98]:
```suggestion
the `mbed_app.json` file which exists in each sample project directory.
```

File: docs/guides/mbedos_platform_overview.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+The Mbed-OS platform makes use of the following build systems to generate
+ninja build scripts:
+
+-   GN - Used by the CHIP project in majority of cases.
+-   CMake (+ mbed-tools) - Used by the Mbed-OS user application and other
+    components within Mbed ecosystem.
+
+As a result, CHIP's stack and platform modules are built with GN and the output
+is used to generate the library file. The
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:110]:
```suggestion
CHIP's stack and platform modules are built with GN and output
a library file. The
```

**Review by pan- — CHANGES_REQUESTED** [review]
Thanks for the intro Lukasz, can you replace CHIP with `Matter` ?

File: docs/guides/mbedos_platform_overview.md (hunk 3 above)
Comment (lukgni) [reply docs/guides/mbedos_platform_overview.md:83]: We decided to not add documentation for `shell` and `all-clusters-app` examples so there is no README.md files there.

Comment (pan-) [reply docs/guides/mbedos_platform_overview.md:83]: Sure but can you link to the application in the source tree ?

File: docs/guides/mbedos_platform_overview.md [hunk 7]
```diff
@@ -106,10 +105,9 @@ ninja build scripts:
 -   CMake (+ mbed-tools) - Used by the Mbed-OS user application and other
     components within Mbed ecosystem.
 
-As a result, Matter's stack and platform modules are built with GN and the output
-is used to generate the library file. The
-application, Mbed-OS and target specific libraries are built with CMake and the Matter
-library file is imported during the compilation process.
+CHIP's stack and platform modules are built with GN and output a library file. The application,
```
Comment (pan-) [inline docs/guides/mbedos_platform_overview.md:108]: Please change CHIP's into Matter's.

File: docs/guides/mbedos_platform_overview.md (hunk 3 above)
Comment (lukgni) [reply docs/guides/mbedos_platform_overview.md]: Done.

File: docs/guides/mbedos_platform_overview.md (hunk 7 above)
Comment (lukgni) [reply docs/guides/mbedos_platform_overview.md]: Done.

**Review by pan- — APPROVED** [review]

**Comment by lukgni** [comment]
Hm.. It looks like this PR has been merged into ` ARMmbed:test-ci-integration` branch.
Maybe I didn't noticed that default branch was changed (for ci testing) when I was creating this PR and that why `test-ci-integration` is used as a base.
@pan- Should we move this change to `development` branch?

**Comment by pan-** [comment]
Yes please move it into development branch.

**Pull request merged**